src/Target.cpp

```diff
@@ -132,8 +132,85 @@ string get_env(const char *name) {
     }
 #endif
 }
+
+const std::map<std::string, Target::OS> os_name_map = {
+    {"os_unknown", Target::OSUnknown},
+    {"linux", Target::Linux},
+    {"windows", Target::Windows},
+    {"osx", Target::OSX},
+    {"android", Target::Android},
+    {"ios", Target::IOS},
+    {"nacl", Target::NaCl},
+};
+
+bool lookup_os(const std::string &tok, Target::OS &result) {
+    auto os_iter = os_name_map.find(tok);
+    if (os_iter != os_name_map.end()) {
+        result = os_iter->second;
+        return true;
+    }
+    return false;
+}
+
+const std::map<std::string, Target::Arch> arch_name_map = {
+    {"arch_unknown", Target::ArchUnknown},
+    {"x86", Target::X86},
+    {"arm", Target::ARM},
+    {"pnacl", Target::PNaCl},
+    {"mips", Target::MIPS},
+};
+
+bool lookup_arch(const std::string &tok, Target::Arch &result) {
+    auto arch_iter = arch_name_map.find(tok);
+    if (arch_iter != arch_name_map.end()) {
+        result = arch_iter->second;
+        return true;
+    }
+    return false;
+}
+
+const std::map<std::string, Target::Feature> feature_name_map = {
```

Comment:
Does this keep the same canonical-target-strings as we have now? If not, code downstream might need changing, (Also probably worth pointing out that we must use map rather than unordered_map to ensure target strings are stable.)